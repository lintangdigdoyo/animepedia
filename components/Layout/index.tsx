import Head from "next/head";

import style from "styles/components/Layout.module.scss";
import Navbar from "components/Navbar";

interface LayoutProps {
  children: React.ReactNode;
}

const Layout = ({ children }: LayoutProps) => {
  return (
    <>
      <Head>
        <title>AnimePedia</title>
        <meta name="description" content="Find your favorite anime" />
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <Navbar />
      <main>{children}</main>
      <footer className={style.layout__footer}>
        Â© 2023 Made by Lintang Digdoyo
      </footer>
    </>
  );
};

export default Layout;
