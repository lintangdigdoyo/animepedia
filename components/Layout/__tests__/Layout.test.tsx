import { render } from "@testing-library/react";
import "@testing-library/jest-dom";

import Layout from "..";

describe("Layout component", () => {
  it("should renders layout correctly", () => {
    const props = {
      children: <>children</>,
    };

    const { container } = render(<Layout {...props} />);
    expect(container).toMatchSnapshot();
  });
});
