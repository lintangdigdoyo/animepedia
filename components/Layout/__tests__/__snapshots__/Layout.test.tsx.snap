// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component should renders layout correctly 1`] = `
<div>
  <header
    class="header"
  >
    <nav
      class="nav"
    >
      <ul
        class="nav__list"
      >
        <li>
          <a
            class="nav__logo"
            data-testid="logo"
            href="/"
          >
            AnimePedia
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main>
    children
  </main>
  <footer
    class="layout__footer"
  >
    Â© 2023 Made by Lintang Digdoyo
  </footer>
</div>
`;
